### DISCUSSION TITLE
Changed Tesla account email:  can no longer login

### DISCUSSION THREAD

**Discussion by JonMurphy** [teslamate-org/teslamate#3618] [Q&A] [answered]
Hello - I am a little lost and I am hoping you can point me in the right direction.

I needed to change my Tesla account email address.  And this has caused LOTS of issues.  One of the issues is related to TeslaMate.

I logged out of **Auth for Tesla** and entered my new Tesla ID/password. I received new API codes and entered those.  But that did not work.

The TeslaMate header turns green and states "Signed In".  After a few minutes the screen changed to this.  It acts like I did not login.

![Screenshot 2024-01-17 at 9 17 49 PM](https://github.com/teslamate-org/teslamate/assets/15616372/4ae851ab-d31e-4b0e-a802-450eaecd2085)

**Comment by cwanja** [comment]
Log out and log in with the new credentialed (e.g. new email address) tokens.

**Comment by JonMurphy** [comment]
I fat fingered something and sent the post before I was done typing.

**Reply by cwanja** [reply]
What happens if you wake the car up via the Tesla app?

**Reply by JonMurphy** [reply]
It was awake before.  I just woke it up again and it still looks the same.  Does it take awhile for things to update?  Do I need to prop open a door for a few minutes to keep it awake?

**Reply by cwanja** [reply]
No, it should wake up and TeslaMate refreshes to start pulling data. 

What do the TeslaMate logs show?

**Reply by JonMurphy** [reply]
I got the right command, but how do I search for today and not the previous 6 months of logs?  There is LOTS of logs

**Reply by JonMurphy** [reply]
Here is what I see.  It repeats over and over:
```
teslamate-teslamate-1 | 2024-01-18 03:33:23.524 [info] Starting logger for 'Sparky II'
teslamate-teslamate-1 | 2024-01-18 03:33:24.048 car_id=1 [info] Start / :online
teslamate-teslamate-1 | 2024-01-18 03:33:24.053 car_id=1 [info] Logged missing software update: 2023.44.30.8 06f534d46010
teslamate-teslamate-1 | 2024-01-18 03:33:24.062 [error] GenStateMachine :"1" terminating
teslamate-teslamate-1 | ** (MatchError) no match of right hand side value: {:error, #Ecto.Changeset<action: :insert, changes: %{date: ~U[2024-01-18 03:33:23.933000Z], power: 0, battery_heater_on: false, battery_level: 80, usable_battery_level: 79, battery_heater: true, driver_temp_setting: Decimal.new("22.0"), fan_status: 0, inside_temp: Decimal.new("-0.6"), is_climate_on: false, is_front_defroster_on: false, is_rear_defroster_on: false, outside_temp: Decimal.new("-1.5"), passenger_temp_setting: Decimal.new("22.0"), odometer: 2631.628126, tpms_pressure_fl: Decimal.new("2.55"), tpms_pressure_fr: Decimal.new("2.6"), tpms_pressure_rl: Decimal.new("2.5"), tpms_pressure_rr: Decimal.new("2.6"), ideal_battery_range_km: Decimal.new("417.35"), rated_battery_range_km: Decimal.new("417.35"), est_battery_range_km: Decimal.new("243.11")}, errors: [latitude: {"can't be blank", [validation: :required]}, longitude: {"can't be blank", [validation: :required]}], data: #TeslaMate.Log.Position<>, valid?: false>}
teslamate-teslamate-1 |     (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:721: anonymous fn/4 in TeslaMate.Vehicles.Vehicle.handle_event/4
teslamate-teslamate-1 |     (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:1203: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
teslamate-teslamate-1 |     (db_connection 2.5.0) lib/db_connection.ex:1630: DBConnection.run_transaction/4
teslamate-teslamate-1 |     (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:713: TeslaMate.Vehicles.Vehicle.handle_event/4
teslamate-teslamate-1 |     (stdlib 5.0.1) gen_statem.erl:1379: :gen_statem.loop_state_callback/11
teslamate-teslamate-1 |     (stdlib 5.0.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
teslamate-teslamate-1 | 2024-01-18 03:33:24.569 car_id=1 [info] Start / :online
teslamate-teslamate-1 | 2024-01-18 03:33:24.573 car_id=1 [info] Logged missing software update: 2023.44.30.8 06f534d46010
teslamate-teslamate-1 | 2024-01-18 03:33:24.582 [error] GenStateMachine :"1" terminating
teslamate-teslamate-1 | ** (MatchError) no match of right hand side value: {:error, #Ecto.Changeset<action: :insert, changes: %{date: ~U[2024-01-18 03:33:24.475000Z], power: 0, battery_heater_on: false, battery_level: 80, usable_battery_level: 79, battery_heater: true, driver_temp_setting: Decimal.new("22.0"), fan_status: 0, inside_temp: Decimal.new("-0.6"), is_climate_on: false, is_front_defroster_on: false, is_rear_defroster_on: false, outside_temp: Decimal.new("-1.5"), passenger_temp_setting: Decimal.new("22.0"), odometer: 2631.628126, tpms_pressure_fl: Decimal.new("2.55"), tpms_pressure_fr: Decimal.new("2.6"), tpms_pressure_rl: Decimal.new("2.5"), tpms_pressure_rr: Decimal.new("2.6"), ideal_battery_range_km: Decimal.new("417.35"), rated_battery_range_km: Decimal.new("417.35"), est_battery_range_km: Decimal.new("243.11")}, errors: [latitude: {"can't be blank", [validation: :required]}, longitude: {"can't be blank", [validation: :required]}], data: #TeslaMate.Log.Position<>, valid?: false>}
teslamate-teslamate-1 |     (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:721: anonymous fn/4 in TeslaMate.Vehicles.Vehicle.handle_event/4
teslamate-teslamate-1 |     (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:1203: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
teslamate-teslamate-1 |     (db_connection 2.5.0) lib/db_connection.ex:1630: DBConnection.run_transaction/4
teslamate-teslamate-1 |     (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:713: TeslaMate.Vehicles.Vehicle.handle_event/4
teslamate-teslamate-1 |     (stdlib 5.0.1) gen_statem.erl:1379: :gen_statem.loop_state_callback/11
teslamate-teslamate-1 |     (stdlib 5.0.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
teslamate-teslamate-1 | 2024-01-18 03:33:24.673 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/37xxxxxxxxx447 -> 429 (84.402 ms)
teslamate-teslamate-1 | 2024-01-18 03:33:24.673 [warning] TeslaApi.Error / "Retry later\n"
teslamate-teslamate-1 | 2024-01-18 03:33:24.673 car_id=1 [error] Error / :unknown
teslamate-teslamate-1 | 2024-01-18 03:33:54.781 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/37xxxxxxxxxx447 -> 429 (105.218 ms)
teslamate-teslamate-1 | 2024-01-18 03:33:54.781 [warning] TeslaApi.Error / "Retry later\n"
teslamate-teslamate-1 | 2024-01-18 03:33:54.782 car_id=1 [error] Error / :unknown
```

**Reply by cwanja** [reply]
What version of TeslaMate are you running?

**Reply by JonMurphy** [reply]
for what it is worth...

- Teslamate Version	1.27.3
- I changed email on December 18.
   - Teslamate looks like it stopped getting data on December 23.
- this is running on an RPi 3B on bookworm

**Reply by cwanja** [reply, marked as answer]
> for what it is worth...
> 
> 
> 
> - Teslamate Version	1.27.3
> 
> - I changed email on December 18.
> 
>    - Teslamate looks like it stopped getting data on December 23.
> 
> - this is running on an RPi 3B on bookworm

Upgrade TeslaMate.

**Reply by JonMurphy** [reply]
Wow - I am way behind.  I missed lots of updates.
Thank you!

**Reply by JonMurphy** [reply]
I am now updated to Version 1.28.2 and I can see my old graphs from December.  THANK YOU!!

I woke the car, but no graph updates yet.  I still see last month's mileage and last month's version in the graphs.  The Home page is correct.

I remember reading about Tesla Third Party apps.  Does that affect Teslamate?

**Reply by cwanja** [reply]
> I am now updated to Version 1.28.2 and I can see my old graphs from December.  THANK YOU!!

Glad you are all set. Highly recommend subscribing to releases. 

> I woke the car, but no graph updates yet.  I still see last month's mileage and last month's version in the graphs.  The Home page is correct.

There is no way to backfill the data. There will just be a gap.

> I remember reading about Tesla Third Party apps.  Does that affect Teslamate?

To be determined. Not clear what Tesla will ultimately do.

**Reply by JonMurphy** [reply]
I wasn't worried about the blanks,  I was worried about receiving new data.  But once I drove the car, there was suddenly new data.  Success!!

> Highly recommend subscribing to releases.

On the bottom of the teslamate **Home** > **Settings** page I thought there used to be a "new update available" notice.  That is what I look for.

I enabled the teslamate **GitHub** > **Watch** > **Custom** > **Releases**.  Hopefully that will work for getting new releases info.  Now I've got to go read version 1.28 release notes so I understand what is new!

Thanks again!
Jon

**Reply by JonMurphy** [reply]
PS - The low tire pressure icon appeared on my TeslaMate screen this morning.  Excellent idea!  I went outside and filled up all of my tires.  I like it!
